src/nemo_safe_synthesizer/llm/utils.py

```
@@ -532,16 +535,62 @@ def _get_auto_tokenizer(
+            str(model_name_or_path),
+        )
+    return cast(PreTrainedTokenizer, tokenizer)
+
+
+def get_device_name() -> str:
+    """Get the name of the current device (first index). Returns 'undefined' if the device is not available."""
+    try:
+        import torch
+
+        return torch.cuda.get_device_properties(0).name
+    except Exception:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)